### ISSUE TITLE
Re-exports do not allow item-style allow(clippy::module_name_repetitions) attributes

### ISSUE THREAD

**Issue by lopopolo** [rust-lang/rust-clippy#7511]
Lint name: `module_name_repetitions`, `useless_attribute`.

I tried this code:

```rust
#![warn(clippy::all)]
#![warn(clippy::pedantic)]

extern crate regex; // 1.5.4

pub mod capture {
    pub use regex::CaptureNames;
}

pub mod split {
    #[allow(clippy::module_name_repetitions)]
    pub use regex::SplitN;
}

pub mod sub {
    #![allow(clippy::module_name_repetitions)]

    pub use regex::SubCaptureMatches;
}

fn main() {}
```

https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=773e01d26c21788a14417433d240e745

I expected to see this happen: Only the `CaptureNames` re-export triggers a lint warning.

Instead, this happened: `SplitN` triggers a useless attribute warning, which differs from the behavior of applying the `allow(clippy:: module_name_repetitions)` pragma to an item.

```
    Checking playground v0.0.1 (/playground)
warning: item name starts with its containing module's name
 --> src/main.rs:7:5
  |
7 |     pub use regex::CaptureNames;
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
note: the lint level is defined here
 --> src/main.rs:2:9
  |
2 | #![warn(clippy::pedantic)]
  |         ^^^^^^^^^^^^^^^^
  = note: `#[warn(clippy::module_name_repetitions)]` implied by `#[warn(clippy::pedantic)]`
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#module_name_repetitions

warning: useless lint attribute
  --> src/main.rs:11:5
   |
11 |     #[allow(clippy::module_name_repetitions)]
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: if you just forgot a `!`, use: `#![allow(clippy::module_name_repetitions)]`
   |
note: the lint level is defined here
  --> src/main.rs:1:9
   |
1  | #![warn(clippy::all)]
   |         ^^^^^^^^^^^
   = note: `#[warn(clippy::useless_attribute)]` implied by `#[warn(clippy::all)]`
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#useless_attribute

warning: `playground` (bin "playground") generated 2 warnings
    Finished dev [unoptimized + debuginfo] target(s) in 0.53s
```

### Meta

Rust 1.54.0 in the playground.
